src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs

```diff
@@ -537,6 +537,58 @@ public async Task POST_ServerAbort_ClientReceivesAbort(HttpProtocols protocol)
         }
     }
 
+    [ConditionalFact]
+    [MsQuicSupported]
+    public async Task POST_ServerAbortAfterWrite_ClientReceivesAbort()
+    {
+        // Arrange
+        var builder = CreateHostBuilder(async context =>
+        {
+            Logger.LogInformation("Server writing content.");
+            await context.Response.Body.WriteAsync(new byte[16]);
+
+            // Note that there is a race here on what is sent before the abort is processed.
+            // Abort may happen before or after response headers have been sent.
+            Logger.LogInformation("Server aborting.");
+            context.Abort();
+        }, protocol: HttpProtocols.Http3);
+
+        using (var host = builder.Build())
+        using (var client = HttpHelpers.CreateClient())
+        {
+            await host.StartAsync().DefaultTimeout();
+
+            for (var i = 0; i < 1000; i++)
```

Comment:
How long does this test take? I don't think tests that run on every CI run and every developer's machines are the right place to do stress testing.

Even if this only takes a second, the road to a long-running test suite are tests like these that are needlessly repetitive. I brought all of Kestrel's 1000s of tests down from many minutes to under 30 seconds when I first moved most of them in memory. That's also why most of our timeout tests use a fake clock and ticks.

We should get this running on dedicated stress infrastructure.

Reply:
It takes 1.3 seconds.

I lowered requests to 100, and it takes 0.16s.